Redesigned the dashboard for the user organisation page

Spandan-Mishra opened the pull request:
## Proposed Changes
Redesigned the user's organization page which was assigned to me in issue #11016.

- Fixes #11016 
- Moved buttons for editing profile and signing out to the right, added vertical ellipsis.
- Added a banner for users who are signed in as doctors
- Added a tab where users can switch between their facilities, associations and administrations.
- Added more localization keys in en.json

**Video which shows the changes made:**

https://github.com/user-attachments/assets/9057a1cf-3eb4-4bdb-a704-32da714f2269


@ohcnetwork/care-fe-code-reviewers

## Merge Checklist

- [ ] Add specs that demonstrate bug / test a new feature.
- [ ] Update [product documentation](https://docs.ohc.network).
- [x] Ensure that UI text is kept in I18n files.
- [x] Prep screenshot or demo video for changelog entry, and attach it to issue.
- [ ] Request for Peer Reviews
- [x] Completion of QA in Mobile Devices
- [x] Completion of QA in Desktop Devices


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **New Features**
  - Introduced a tabbed dashboard interface that organizes user-related information into sections: Facilities, Associations, and Governance.
  - Enhanced welcome messages with more personalized greetings, including specific messages for doctors.
  - Added new contextual information regarding affiliations and healthcare facilities in the dashboard.
  - Added new entries for displaying messages when no associations or governance are found.
- **Bug Fixes**
  - Improved filtering logic for displaying organizations based on user roles and governance types.
- **Style**
  - Refined the layout and visual presentation across dashboard sections for improved clarity and consistency, including adjustments to text alignment and avatar size.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

Spandan-Mishra committed:
Redesigned the dashboard for the user organisation page

bodhish commented:
Lets rename `administration` to `governance`

Spandan-Mishra commented:
any other changes needed? @bodhish

Spandan-Mishra committed:
Fixed proposed changes in PR

Added UI feedback when lists are empty, added accessibilty attributes and fixed some typos

Spandan-Mishra commented:
@nihal467 should I make the changes mentioned by coderabbit and make another commit?

Jacobjeevan commented:
> @nihal467 should I make the changes mentioned by coderabbit and make another commit?

If the suggestion makes sense yes

yash-learner commented on src/pages/UserDashboard.tsx:
```diff
@@ -19,144 +26,281 @@ export default function UserDashboard() {
+              className="w-full sm:w-auto"
+              asChild
+            >
+              <EllipsisVerticalIcon className="h-4 w-4" />
```
You can use `care icon` here https://care.ohc.network/icons `<CareIcon icon="l-ellipsis-v" />`

yash-learner commented on src/pages/UserDashboard.tsx:
```diff
@@ -19,144 +26,281 @@ export default function UserDashboard() {
+            </section>
+          )}
+
+          {/*Associations Section*/}
```
We can remove comments like these. Use only when necessary

yash-learner reviewed (changes requested):
As per the [Figma design](https://www.figma.com/design/Z93EYKSa1MdBmXndsMBJSQ/Care?node-id=9717-2482&t=cueXLL157eEEV4OE-0), the `Sign out` button is expected to move inside the ` kebab menu` .

cc: @vinutv

Spandan-Mishra committed:
Formatted code better and moved signout under ellipsis

Spandan-Mishra commented:
@nihal467 @Jacobjeevan 
Implemented the changes given by coderabbit. Except moving the tab contents to another component, as it was making the code more complex, and did not reduce code redundancy.

@yash-learner 
Moved Sign Out under the menu, used CareIcon and removed unnecessary comments.

AdityaJ2305 commented:
@Spandan-Mishra , do "Mark as Resolved" with comment or acknowledgement after changes are done or unnecessary ones by coderabbitai

Spandan-Mishra commented:
Resolved all useful as well as unnecessary changes @AdityaJ2305

yash-learner commented on src/pages/UserDashboard.tsx:
```diff
@@ -12,151 +21,301 @@ import useAuthUser, { useAuthContext } from "@/hooks/useAuthUser";
+            >
+              <p className="text-sm text-gray-800 font-normal px-1">
+                {t("dashboard_tab_facilities")}
+              </p>
+              {facilities.length === 0 ? (
+                <div className="text-center py-6 text-gray-500">
+                  <p>{t("no_facilities_found")}</p>
+                </div>
+              ) : (
+                <div
+                  className="grid gap-3 md:gap-4 grid-cols-1 sm:grid-cols-2 lg:grid-cols-3"
+                  data-cy="facility-list"
+                >
+                  {facilities.map((facility) => (
+                    <Link
+                      key={facility.id}
+                      href={`/facility/${facility.id}/overview`}
+                    >
+                      <Card className="transition-all hover:shadow-md hover:border-primary/20">
+                        <CardContent className="flex items-center gap-3 p-3 md:p-4">
+                          <Avatar
+                            name={facility.name}
+                            className="h-12 w-12 md:h-14 md:w-14"
+                          />
+                          <div className="flex-1 min-w-0">
+                            <h3 className="font-medium truncate text-sm md:text-base">
+                              {facility.name}
+                            </h3>
+                            <p className="text-xs md:text-sm text-gray-500 truncate">
+                              {t("view_facility_details")}
+                            </p>
+                          </div>
+                          <ChevronRight className="h-4 w-4 md:h-5 md:w-5 text-gray-500" />
+                        </CardContent>
+                      </Card>
+                    </Link>
+                  ))}
+                </div>
+              )}
+            </section>
```
You can extract this into a separate component, repeating the same type of structure for all three tabs

yash-learner reviewed (changes requested):
![image](https://github.com/user-attachments/assets/6eddcb7b-0dc7-4a12-aa33-a1fdc516260f)


I think we can make this card better by moving `Admin Dashboard` and `Edit profile` button as well inside the `kekab menu` for  mobile devices 


---

Also align the Avatar and text below it properly on mobile devices

---


I think we can change the `My Facilities` to `Facilities` cc: @ohcnetwork/care-fe-code-reviewers

yash-learner commented:
@Spandan-Mishra 

Adding for context. here is the discussion on slack regarding UI/UX of 
 card on mobile https://rebuildearth.slack.com/archives/C080225HC6B/p1741346917485939

Spandan-Mishra committed:
Fixed all proposed changes

Made the design responsive for mobile and upda
ted card design for mobiles. Introduced TabContent Component to reduce code redundancy

Spandan-Mishra commented:
made the card mobile responsive and reduced code redundancy
@yash-learner @Jacobjeevan

Jacobjeevan commented on public/locale/en.json:
```diff
@@ -2575,8 +2580,10 @@
+  "welcome_back": "Welcome back,",
   "welcome_back_to_hospital_dashboard": "Welcome back to the overview ",
+  "welcome_dr": "Dr. {{name}}",
+  "welcome_dr_banner": "You are logged in as a doctor!",
```
what about other user types? 🤔 

Let's change it to accept the user type (similar welcome_dr).

Spandan-Mishra replied:
The user_type field either had 'doctor' value or none. So what should I implement?

Spandan-Mishra replied:
the welcome message will display for all kinds of users. doctors have a different format
the banner will be visible to only those with a doctor role.

rithviknishad replied:
there's plans to add prefix field for users to include this case. let's not render based on user_type. user_type is likely to be removed soon in favour of the new RBAC.

Jacobjeevan commented on src/pages/UserDashboard.tsx:
```diff
@@ -12,151 +21,319 @@ import useAuthUser, { useAuthContext } from "@/hooks/useAuthUser";
+  [key: string]: any;
+};
+
+type RenderLinkFunction = (item: Item) => JSX.Element;
```
Let's inline this instead of defining a separate type.

Jacobjeevan commented on src/pages/UserDashboard.tsx:
```diff
@@ -12,151 +21,319 @@ import useAuthUser, { useAuthContext } from "@/hooks/useAuthUser";
+                  <LogOut className="h-4 w-4" />
+                  {t("sign_out")}
+                </DropdownMenuItem>
+                <DropdownMenuItem className="cursor-pointer flex sm:hidden items-center gap-2 text-xs w-full sm:w-auto">
```
let's move edit profile above sign out (don't forget to adjust the visibility attribute once you remove button outside of the dropdown menu).

Jacobjeevan commented on src/pages/UserDashboard.tsx:
```diff
@@ -12,151 +21,319 @@ import useAuthUser, { useAuthContext } from "@/hooks/useAuthUser";
+                </Button>
+              </DropdownMenuTrigger>
+              <DropdownMenuContent align="end" className="w-40">
+                {user.is_superuser && (
```
let's remove admin dashboard in drop down, redundant anyways.

Spandan-Mishra replied:
This will only be visible in mobile view. In desktop mode it is hidden. Should I change it or let it be?

Jacobjeevan commented on src/pages/UserDashboard.tsx:
```diff
@@ -12,151 +21,319 @@ import useAuthUser, { useAuthContext } from "@/hooks/useAuthUser";
-                <User2Icon className="h-4 w-4" />
-                {t("admin_dashboard")}
+                <SquarePen className="h-4 w-4" />
+                {t("edit_profile")}
```
let's also remove the edit profile button outside the kebab menu

Spandan-Mishra replied:
So which buttons should the kebab menu have in desktop and mobile views? Please do specify

Spandan-Mishra committed:
Minor changes in profile card

Spandan-Mishra committed:
Admin dashboard restricted to superusers

Spandan-Mishra commented:
@Jacobjeevan made all the changes which were requested.
this is how it looks rn:


https://github.com/user-attachments/assets/bd4c1299-3c02-43e7-8ff0-736bae5b40a0

Jacobjeevan commented on src/pages/UserDashboard.tsx:
```diff
@@ -12,151 +21,303 @@ import useAuthUser, { useAuthContext } from "@/hooks/useAuthUser";
+                  {t("welcome_back")}
+                </p>
+                <h1 className="text-xl md:text-2xl">
+                  {user.user_type === "doctor"
```
Let's remove the banner for now, as Rithvik said, prefixes will be added soonish, we can add the banner then. Make sure to remove the key from en.json.

Jacobjeevan replied:
<img width="461" alt="image" src="https://github.com/user-attachments/assets/ffeeefce-c78c-455b-9815-e6d0bb6cc2c8" />

Adjust for smaller devices as well.

Spandan-Mishra replied:
Custom breakpoint xs needs to be added to make flex-col for a screen of that size. Should I do it?

Jacobjeevan replied:
No, that's not the solution; you would need to adjust the entire page, it's not just the cards that are overflowing.

Spandan-Mishra replied:
Fixed this and removed the banner

Spandan-Mishra commented:
@Jacobjeevan I'll remove the banner. Should I keep the welcome_dr message or remove it and make it the same for all users?

PS: Please do let me know when you want to implement the banner 😅. I'd like to work on it

Spandan-Mishra committed:
Removed overflow and responsiveness added for smaller devices

Spandan-Mishra committed:
Removed doctor banner message

Jacobjeevan commented on src/pages/UserDashboard.tsx:
```diff
@@ -148,18 +148,13 @@ export default function UserDashboard() {
         {/* Tabs Headings */}
         <div
-          className="flex border-b border-gray-200"
+          className="flex flex-wrap justify-center border-b border-gray-200"
```
don't need justify-center here.

Spandan-Mishra replied:
I'll look into it 👍🏻

yash-learner reviewed (changes requested):
**Expected design for mobile based on figma**

![image](https://github.com/user-attachments/assets/dd133be8-78df-4c8f-93f0-55afcd7a026a)



https://www.figma.com/design/Z93EYKSa1MdBmXndsMBJSQ/Care?node-id=10481-1260&t=fe2eOLdMvNGFjTlE-0



**current** 



![image](https://github.com/user-attachments/assets/4afeeaf5-0374-4d10-bac4-0e56d49697fa)


----


![image](https://github.com/user-attachments/assets/c15a44e5-8558-480f-97fa-34591ddc899a)


**Tab names:** Add horizontal overflow scroll  and make text size as `text-sm` for mobiles.

Spandan-Mishra commented:
@yash-learner I'll work on it.
Options like admin dashboard, edit profile etc., will still be in the kebab menu right? (figma file has a different design)

Spandan-Mishra committed:
Mobile reponsiveness changes

Spandan-Mishra commented:
@Jacobjeevan @yash-learner made the changes.
For smaller screens, a swipe or arrow feature for tab names could be added, current one shows little overflow at the end (manageable)

Jacobjeevan committed:
adjust types

yash-learner commented on src/pages/UserDashboard.tsx:
```diff
@@ -19,144 +43,276 @@ export default function UserDashboard() {
+              tabItems: associations,
+              emptyMessage: t("no_associations_found"),
+              description: t("dashboard_tab_associations"),
+              renderChild: (association) => (
+                <Link
+                  key={association.id}
+                  href={`/organization/${association.id}`}
+                >
+                  <Card className="transition-all hover:shadow-md hover:border-primary/20">
+                    <CardContent className="flex items-center gap-3 p-3 md:p-4">
+                      <Avatar
+                        name={association.name}
+                        className="h-12 w-12 md:h-14 md:w-14"
+                      />
+                      <div className="flex-1 min-w-0">
+                        <h3 className="font-medium truncate text-sm md:text-base">
+                          {association.name}
+                        </h3>
+                        <p className="text-xs md:text-sm text-gray-500 truncate">
+                          {"org_type" in association &&
+                            getOrgLabel(
+                              association.org_type,
+                              association.metadata,
+                            )}
+                        </p>
+                      </div>
+                      <ChevronRight className="h-4 w-4 md:h-5 md:w-5 text-gray-500" />
+                    </CardContent>
+                  </Card>
+                </Link>
+              ),
```
We can extract this into a separate component and reuse for all three tabs, structure is same, the only difference is data we are passing

Spandan-Mishra replied:
fixed this @yash-learner

Spandan-Mishra committed:
code redundancy fixed

Spandan-Mishra committed:
Resolve merge conflict in package-lock.json

Spandan-Mishra committed:
Remove unwanted files from PR

Spandan-Mishra commented:
@Jacobjeevan one of the commits caused some kind of merge conflict, how to fixt it?

Spandan-Mishra committed:
Saving local changes before merging

bodhish committed:
Update logs

bodhish reviewed (approved):
LGTM
